examples/cpp_execution/text_inference_using_cpp.py

```diff
@@ -0,0 +1,283 @@
+    return compilation_batch_size, compilation_ctx_len
+
+
+def read_prompts_txt_file(prompts_txt_file_path: str):
+    prompt = []
+    with open(prompts_txt_file_path, "r") as file:
+        for line in file:
+            prompt.append(line.strip())
+    return prompt
+
+
+def get_input_prompts(prompt: str, prompts_txt_file_path: str) -> List[str]:
+    assert (
+        prompt is not None or prompts_txt_file_path is not None
+    ), "Please pass at least one argument either using --prompt or --prompts_txt_file_path"
+    if prompts_txt_file_path is not None:
+        if prompt is not None:
+            logger.warning("Found inputs passed using txt file as well as CLI, taking inputs from given txt file")
+        prompt = read_prompts_txt_file(prompts_txt_file_path)
+    if isinstance(prompt, str):
+        prompt = [prompt]
+    return prompt
+
+
+def fix_prompts(prompt: List[str], batch_size: int, full_batch_size: int = None):
+    """
+    Adjusts the list of prompts to match the required batch size.
+
+    ``Mandatory`` Args:
+        prompt (List[str]): List of input prompts.
+        batch_size (int): The batch size to process at a time.
+
+    ``Optional`` Args:
+        full_batch_size (Optional[int]): The full batch size if different from batch_size.
+
+    Returns:
+        List[str]: Adjusted list of prompts.
+    """
+    exec_batch_size = full_batch_size if full_batch_size is not None else batch_size
+
+    if len(prompt) < exec_batch_size:
+        logger.warning("Number of prompts are less than batch size/full batch size, repeating to required batch size")
+        prompt = (prompt * (exec_batch_size // len(prompt) + 1))[:exec_batch_size]
+    elif full_batch_size is None and len(prompt) % batch_size != 0:
+        logger.warning(
+            "Number of prompts are not multiple of batch size, dropping last incomplete batch from given input prompts"
+        )
+        prompt = prompt[: batch_size * (len(prompt) // batch_size)]
+
+    return prompt
+
+
+def cloud_ai_100_exec_kv(
+    tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
+    qpc_path: str,
+    prompt: Optional[List[str]] = None,
+    prompts_txt_file_path: Optional[str] = None,
+    device_id: Optional[List[int]] = None,
+    generation_len: Optional[int] = None,
+    enable_debug_logs: bool = False,
+    stream: bool = True,
+    full_batch_size: Optional[int] = None,
+):
+    batch_size, ctx_len = get_compilation_dims(qpc_path)
+    prompt: List[str] = get_input_prompts(prompt, prompts_txt_file_path)
+    prompt = fix_prompts(prompt, batch_size, full_batch_size)
+
+    # ********* CPP Calling ********
+    InferenceSetIOBufferExample.generatePrompt(
+        tokenizer, qpc_path, batch_size, ctx_len, prompt, generation_len, device_id
+    )
+
+
+def tokenize_for_prefill(prompt, tokenizer, padded_len):
+    inputs = tokenizer(prompt, return_tensors="np", padding="max_length", max_length=padded_len)
```

Comment:
add these lines above 189,

inputs = tokenizer(prompt, return_tensors="np", padding=True)
padded_len = inputs["input_ids"].shape[1]
num_chunks = -(padded_len // -prefill_seq_len)  # ceil divide without float
padded_len = num_chunks * prefill_seq_len

remove padded_len and add prefill_seq_len to input to this function and return num_chunks as output.